src/dmd/mtype.d

```diff
@@ -4736,6 +4736,7 @@ extern (C++) final class TypeFunction : TypeNext
                              * copytmp.__copyCtor(arg);
                              */
                             auto tmp = new VarDeclaration(arg.loc, tprm, Identifier.generateId("__copytmp"), null);
```

Comment:
There is a more lightweight technique to probe the copy constructor but maybe leave refactoring for a separate PR.
See:
https://github.com/dlang/dmd/blob/664f07c08d4331b53e6fcb419a070c808dc4e71d/src/dmd/clone.d#L103
https://github.com/dlang/dmd/blob/664f07c08d4331b53e6fcb419a070c808dc4e71d/src/dmd/clone.d#L113

Reply:
Yep. Even the temporary generation can be substituted with a simple function call (copyToTemp), but I didn't want to mix any refactorizations in.